src/state/predictions/index.ts

```diff
@@ -439,6 +489,35 @@ export const predictionsSlice = createSlice({
       state.ledgers = merge({}, state.ledgers, action.payload)
     })
 
+    // Get static market data
+    builder.addCase(fetchPredictionData.fulfilled, (state, action) => {
+      const { status, currentEpoch, intervalSeconds, minBetAmount, rounds, claimableStatuses, ledgers } = action.payload
+
+      const allRoundData = merge({}, state.rounds, rounds)
+      let newRounds = pickBy(allRoundData, (value, key) => {
+        return Number(key) > state.currentEpoch - PAST_ROUND_COUNT
+      })
+
+      // If the round has change add a new future round
```

Comment:
Adding new future round after trimming makes the transition correctly. Next fetching cycle the earlist round will be trimmed and it will be handled in here https://github.com/pancakeswap/pancake-frontend/pull/3572/files#diff-f1462bf4b1ea4d9ef14f0ed608ff12ad755d8a05854aace98963662164cd3bb3R43